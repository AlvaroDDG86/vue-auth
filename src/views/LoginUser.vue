<template>
  <div>
    <h1>Login</h1>
    <form class="noauth" @submit.prevent="login">
      <label for="email">Email: </label>
      <input id="email" v-model="email" name="email" type="text">
      <label for="password">Password: </label>
      <input id="password" v-model="password" name="password" type="password">
      <button name="button">
        Login
      </button>
      <p v-if="error">
        {{ error }}
      </p>
    </form>
    <p>
      Don't have an account?
      <router-link to="/register">
        Register
      </router-link>
    </p>
  </div>
</template>

<script>
export default {
  data () {
    return {
      email: '',
      password: '',
      error: ''
    }
  },
  methods: {
    login () {
      this.error = ''
      this.$store.dispatch('login', {
        email: this.email,
        password: this.password
      })
        .then(() => {
          this.$router.push({ name: 'dashboard' })
        })
        .catch(error => {
          this.error = error.response.data.error
        })
    }
  }
}
</script>

<style lang="scss" scoped>
</style>
