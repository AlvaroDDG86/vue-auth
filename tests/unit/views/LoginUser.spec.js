import { shallowMount, createLocalVue } from '@vue/test-utils'
import LoginUser from '@/views/LoginUser'
import Vuex from 'vuex'

const localVue = createLocalVue()
localVue.use(Vuex)

const mockRouter = {
  push: jest.fn()
}

describe('RegisterUser', () => {
  let wrapper
  let actions
  let store
  beforeEach(() => {
    actions = {
      login: jest.fn(() => Promise.resolve(''))
    }
    store = new Vuex.Store({
      actions
    })
    wrapper = shallowMount(LoginUser, {
      store,
      localVue,
      mocks: {
        $router: mockRouter
      }
    })
  })
  test('should match with snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  test('should register and navigate to dashboard', async () => {
    await wrapper.find('[name="email"]').setValue('alice@facebook.com')
    await wrapper.find('[name="password"]').setValue('111111')
    await wrapper.find('form').trigger('submit')
    expect(mockRouter.push).toHaveBeenCalled()
    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'dashboard' })
  })
})
