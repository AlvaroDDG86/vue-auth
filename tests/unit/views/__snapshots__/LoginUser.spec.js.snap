// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterUser should match with snapshot 1`] = `
<div>
  <form><label for="email">Email: </label> <input id="email" name="email" type="text"> <label for="password">Password: </label> <input id="password" name="password" type="password"> <button name="button">
      Login
    </button></form>
</div>
`;
